src/Guzzle/MiddlewareFactory.php

```diff
@@ -53,7 +53,22 @@ public function retry( $delay = true ) {
 	 * @return callable
 	 */
 	private function getRetryDelay() {
-		return function( $numberOfRetries ) {
+		return function( $numberOfRetries, Response $response = null ) {
+			// The $response argument is only passed as of Guzzle 6.2.2.
+			if( $response !== null ) {
+				// Retry-After may be a number of seonds or an absolute date.
+				$retryAfter = $response->getHeaderLine( 'Retry-After' );
+
+				if( is_numeric( $retryAfter ) ) {
+					return 1000 * $retryAfter;
+				}
+
+				if( $retryAfter ) {
+					$seconds = strtotime( $retryAfter ) - time();
```

Comment:
Is there some documentation that states that the header may contain a date / timestamp rather than a value in seconds?
I can't see it on the wiki page and I imagine including code like this will lead to some odd wait times if a date is returned due to timezones.